### PR TITLE
D-002 Wallet password improvements: follow NIST guidelines

### PR DESCRIPTION
**Pull request by LendiHop**

### CODE REVIEW

File: src/form/form-password-input.tsx [hunk 1]
```diff
@@ -1,19 +1,54 @@
...
+export interface PasswordValidation {
+  minChar: boolean;
+  cases: boolean;
+  number: boolean;
+  specialChar: boolean;
+}
+
 interface Props extends Pick<StyledPasswordInputProps, 'testID'> {
   name: string;
+  isNewPassword?: boolean;
```
Comment (letier3110) [inline src/form/form-password-input.tsx:27]: strange naming, its used not everywhere?

Comment (LendiHop) [reply src/form/form-password-input.tsx:27]: yes. Only for when user enters new wallet password

**Review by letier3110 — APPROVED** [review]

File: src/form/form-password-input.tsx [hunk 2]
```diff
@@ -1,19 +1,54 @@
...
+
+  const handleChange = (text: string) => {
+    helpers.setValue(text);
+    setPasswordValidation({
+      minChar: text.length >= MIN_PASSWORD_LENGTH,
+      cases: uppercaseLowercaseMixtureRegx.test(text),
+      number: lettersNumbersMixtureRegx.test(text),
+      specialChar: specialCharacterRegx.test(text)
+    });
+  };
```
Comment (unixvb) [inline src/form/form-password-input.tsx:50]: The same mistake that was done on extension.  
This functionality is not related to shared password input.  extract this logic out of here.

File: src/components/password-strength-indicator/password-strength-indicator.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { PasswordValidation } from '../../form/form-password-input';
+import { isDefined } from '../../utils/is-defined';
+import { PasswordStrengthIndicatorItem } from './password-strength-indicator-item/password-strength-indicator-item';
+
+interface PasswordStrengthIndicatorProps {
+  validation: PasswordValidation;
+  isError?: boolean;
+}
+
+interface ValidationMessage {
```
Comment (unixvb) [inline src/components/password-strength-indicator/password-strength-indicator.tsx:13]: mb just reuse `PasswordStrengthIndicatorItemProps`?

**Review by unixvb — APPROVED** [review]
